crates/agent/src/api/update_l2_reporting.rs

```diff
@@ -174,41 +173,32 @@ export class Derivation extends Types.IDerivation {"#
         ..Default::default()
     };
 
-    let pub_id = id_generator.lock().unwrap().next();
     let logs_token = uuid::Uuid::new_v4();
-
-    let uncommitted = publisher
-        .build(
-            user_id,
-            pub_id,
-            Some(format!("publication for updating L2 reporting")),
-            draft,
-            logs_token,
-            &default_data_plane,
-        )
-        .await?;
-
-    if uncommitted.has_errors() {
-        for err in uncommitted.output.errors() {
-            tracing::error!(scope=%err.scope, err=format!("{:#}", err.error), "data-plane-template build error")
+    let publication = DraftPublication {
+        user_id,
+        logs_token,
+        draft,
+        dry_run,
+        detail: Some(format!("publication for updating L2 reporting")),
+        default_data_plane_name: Some(default_data_plane.clone()),
+        // TODO: should we verify user authz for updating L2 reporting?
```

Comment:
Same story, it's done first-thing by the routine, unless I'm not understanding the question.
I suppose a difference here is that the user could be an admin of `ops/` but  _not_ be an admin of `ops.us-central1/` and then it would fail? Meh :shrug: